src/databricks/labs/ucx/source_code/queries.py

```diff
@@ -156,20 +198,34 @@ def lint_query(self, query: LegacyQuery) -> Iterable[QueryProblem]:
                 message=advice.message,
             )
 
-    @classmethod
-    def collect_dfsas_from_query(cls, dashboard_id: str, query: LegacyQuery) -> Iterable[DirectFsAccess]:
+    def collect_dfsas_from_query(self, dashboard_id: str, query: LegacyQuery) -> Iterable[DirectFsAccess]:
         if query.query is None:
             return
-        linter = DirectFsAccessSqlLinter()
+        ctx = LinterContext(self._migration_index, CurrentSessionState())
+        collector = ctx.dfsa_collector(Language.SQL)
         source_id = f"{dashboard_id}/{query.id}"
         source_name = query.name or "<anonymous>"
-        source_timestamp = cls._read_timestamp(query.updated_at)
+        source_timestamp = self._read_timestamp(query.updated_at)
         source_lineage = [LineageAtom(object_type="QUERY", object_id=source_id, other={"name": source_name})]
-        for dfsa in linter.collect_dfsas(query.query):
+        for dfsa in collector.collect_dfsas(query.query):
             yield DirectFsAccess(**asdict(dfsa)).replace_source(
                 source_id=source_id, source_timestamp=source_timestamp, source_lineage=source_lineage
             )
 
+    def collect_used_tables_from_query(self, dashboard_id: str, query: LegacyQuery) -> Iterable[UsedTable]:
+        if query.query is None:
+            return
+        ctx = LinterContext(self._migration_index, CurrentSessionState())
+        collector = ctx.tables_collector(Language.SQL)
+        source_id = f"{dashboard_id}/{query.id}"
+        source_name = query.name or "<anonymous>"
+        source_timestamp = self._read_timestamp(query.updated_at)
+        source_lineage = [LineageAtom(object_type="QUERY", object_id=source_id, other={"name": source_name})]
+        for table in collector.collect_tables(query.query):
+            yield UsedTable(**asdict(table)).replace_source(
```

Comment:
```suggestion
            yield UsedTable(**asdict(table)).replace_source(
```
use dataclasses.replace instead

Reply:
simplified - replace_source uses dataclesses.replace